### PR TITLE
Update json docs

### PR DESCRIPTION
**Pull request by syoung-smallwisdom**
With Xcode 12 and SPM, I think we should be able to add a script to build the JSON automatically but for now, this is a manual mention for building documentation.

### CODE REVIEW

File: Research/Research/AnswerTypeObjects.swift [hunk 1]
```diff
@@ -39,7 +39,7 @@ public final class AnswerTypeSerializer : AbstractPolymorphicSerializer, Polymor
         """
         `AnswerType` is used to allow carrying additional information about the properties of a
         JSON-encoded `AnswerResult`.
-        """
+        """.replacingOccurrences(of: "\n", with: " ").replacingOccurrences(of: "  ", with: "\n")
```
Comment (Erin-Mounts) [inline Research/Research/AnswerTypeObjects.swift:42]: What's up with these string replacements? It's not obvious to me what problem this is solving.

File: Research/Research/AssessmentTaskObject.swift [hunk 2]
```diff
@@ -61,12 +61,12 @@ extension RSDTaskType : DocumentableStringLiteral {
...
 
-public final class TaskSerializer : AbstractPolymorphicSerializer, PolymorphicSerializer {
+public final class TaskSerializer : IdentifiableInterfaceSerializer, PolymorphicSerializer {
     public var documentDescription: String? {
         """
-        `RSDTask` is the interface for running a task. It includes information about how to
+        `Task` is the interface for running a task. It includes information about how to
         calculate progress, validation, and the order of display for the steps.
-        """
+        """.replacingOccurrences(of: "\n", with: " ").replacingOccurrences(of: "  ", with: "\n")
```
Comment (Erin-Mounts) [inline Research/Research/AssessmentTaskObject.swift:69]: see previous comment about purpose of string replacements

File: Research/Research/InputItemObjects.swift [hunk 3]
```diff
@@ -47,7 +47,7 @@ public final class InputItemSerializer : AbstractPolymorphicSerializer, Polymorp
         In another example, the input item could be a single cell in a list that shows the possible
         choices for a question. In essence, this is akin to a single cell in a table view though
         the actual implementation may differ.
-        """
+        """.replacingOccurrences(of: "\n", with: " ").replacingOccurrences(of: "  ", with: "\n")
```
Comment (Erin-Mounts) [inline Research/Research/InputItemObjects.swift:50]: ditto

File: Research/Research/RSDAsyncActionType.swift [hunk 4]
```diff
@@ -34,18 +34,18 @@
...
+        `AsyncActionConfiguration` defines general configuration for an asynchronous action that
         should be run in the background. Depending upon the parameters and how the action is set up,
         this could be something that is run continuously or else is paused or reset based on a
         timeout interval.
         
         The configuration is intended to be a serializable object and does not call services, record
         data, or anything else. It does include a step identifier that can be used to let the
         task controller know when to trigger the async action.
-        """
+        """.replacingOccurrences(of: "\n", with: " ").replacingOccurrences(of: "  ", with: "\n")
```
Comment (Erin-Mounts) [inline Research/Research/RSDAsyncActionType.swift:48]: ditto

File: Research/Research/RSDColorMappingThemeElementObject.swift [hunk 5]
```diff
@@ -69,10 +69,10 @@ extension RSDColorMappingThemeElementType : DocumentableStringLiteral {
 public final class ColorMappingSerializer : AbstractPolymorphicSerializer, PolymorphicSerializer {
     public var documentDescription: String? {
         """
-        `RSDColorMappingThemeElement` defines the colors to use on a given screen. Typically, this
+        `ColorMappingThemeElement` defines the colors to use on a given screen. Typically, this
         includes the background color for the header or a background color that is applied to the
         full screen.
-        """
+        """.replacingOccurrences(of: "\n", with: " ").replacingOccurrences(of: "  ", with: "\n")
```
Comment (Erin-Mounts) [inline Research/Research/RSDColorMappingThemeElementObject.swift:75]: ditto

File: Research/Research/RSDImageThemeObject.swift [hunk 6]
```diff
@@ -69,8 +69,8 @@ extension RSDImageThemeElementType : DocumentableStringLiteral {
 public final class ImageThemeSerializer : AbstractPolymorphicSerializer, PolymorphicSerializer {
     public var documentDescription: String? {
         """
-        `RSDImageThemeElement` extends the UI step to include an image."
-        """
+        `ImageInfo` extends the UI step to include an image."
+        """.replacingOccurrences(of: "\n", with: " ").replacingOccurrences(of: "  ", with: "\n")
```
Comment (Erin-Mounts) [inline Research/Research/RSDImageThemeObject.swift:73]: ditto

File: Research/Research/RSDResultType.swift [hunk 7]
```diff
@@ -82,13 +82,13 @@ extension RSDResultType : DocumentableStringLiteral {
...
-public final class RSDResultSerializer : AbstractPolymorphicSerializer, PolymorphicSerializer {
+public final class RSDResultSerializer : IdentifiableInterfaceSerializer, PolymorphicSerializer {
     public var documentDescription: String? {
         """
-        `RSDResult` is the base implementation for a result associated with a task, step, or
+        `Result` is the base implementation for a result associated with a task, step, or
         asynchronous action. When running a task, there will be a result of some variety used to
         mark each step in the task.
-        """
+        """.replacingOccurrences(of: "\n", with: " ").replacingOccurrences(of: "  ", with: "\n")
```
Comment (Erin-Mounts) [inline Research/Research/RSDResultType.swift:91]: ditto

File: Research/Research/RSDStepType.swift [hunk 8]
```diff
@@ -139,13 +139,13 @@ extension RSDStepType : DocumentableStringLiteral {
...
-public final class StepSerializer : AbstractPolymorphicSerializer, PolymorphicSerializer {
+public final class StepSerializer : IdentifiableInterfaceSerializer, PolymorphicSerializer {
     public var documentDescription: String? {
         """
-        `RSDStep` is the base protocol for the steps that can compose a task for presentation using
+        `Step` is the base protocol for the steps that can compose a task for presentation using
         a controller appropriate to the device and application. Each `RSDStep` object represents one
         logical piece of data entry, information, or activity in a larger task.
-        """
+        """.replacingOccurrences(of: "\n", with: " ").replacingOccurrences(of: "  ", with: "\n")
```
Comment (Erin-Mounts) [inline Research/Research/RSDStepType.swift:148]: ditto

File: Research/Research/RSDUIActionObject.swift [hunk 9]
```diff
@@ -83,9 +83,9 @@ extension RSDUIActionObjectType : DocumentableStringLiteral {
 public final class ButtonActionSerializer : AbstractPolymorphicSerializer, PolymorphicSerializer {
     public var documentDescription: String? {
         """
-        `RSDUIAction` protocol can be used to customize the title and image displayed for a given
+        `ButtonActionInfo` protocol can be used to customize the title and image displayed for a given
         action of the UI.
-        """
+        """.replacingOccurrences(of: "\n", with: " ").replacingOccurrences(of: "  ", with: "\n")
```
Comment (Erin-Mounts) [inline Research/Research/RSDUIActionObject.swift:88]: ditto

File: Research/Research/RSDViewThemeElementObject.swift [hunk 10]
```diff
@@ -37,9 +37,9 @@ import JsonModel
 public final class ViewThemeSerializer : AbstractPolymorphicSerializer, PolymorphicSerializer {
     public var documentDescription: String? {
         """
-        `RSDViewThemeElement` tells the UI where to find the view controller or fragment to use
+        `ViewThemeElement` tells the UI where to find the view controller or fragment to use
         when instantiating the view.
-        """
+        """.replacingOccurrences(of: "\n", with: " ").replacingOccurrences(of: "  ", with: "\n")
```
Comment (Erin-Mounts) [inline Research/Research/RSDViewThemeElementObject.swift:42]: ditto

File: Research/Research/ResultNodeSerializer.swift [hunk 11]
```diff
@@ -34,9 +34,11 @@
 import Foundation
 import JsonModel
 
-public final class ResultNodeSerializer : AbstractPolymorphicSerializer, PolymorphicSerializer {
+public final class ResultNodeSerializer : IdentifiableInterfaceSerializer, PolymorphicSerializer {
     public var documentDescription: String? {
-        "`ResultNode` is an interface used to allow for a related grouping of questions."
+        """
+        `ResultNode` is an interface used to allow for a related grouping of questions.
+        """.replacingOccurrences(of: "\n", with: " ").replacingOccurrences(of: "  ", with: "\n")
```
Comment (Erin-Mounts) [inline Research/Research/ResultNodeSerializer.swift:41]: ditto

**Review by Erin-Mounts — APPROVED** [review]
One question (repeated in a whole bunch of places), otherwise looks good

File: Research/Research/AnswerTypeObjects.swift (hunk 1 above)
Comment (syoung-smallwisdom) [reply Research/Research/AnswerTypeObjects.swift:42]: Formatting of the string in the JSON. It looks really weird with the newlines.

**Pull request merged**